tools/topology/platform/intel/cml.m4

```diff
@@ -10,4 +10,54 @@ define(`SSP_NAME', `SSP0-Codec')
 undefine(`SSP_MCLK_RATE')
 define(`SSP_MCLK_RATE', `24000000')
 
+# Low Latency playback pipeline 8 on PCM 6 using max 2 channels of s32le.
+# Schedule 1000us deadline on core 0 with priority 0
+# pipe-src pipeline is needed for older SSP config
+ifelse(SOF_ABI_VERSION_3_9_OR_GRT, `0',
+        define(`PIPE_VOLUME_PLAYBACK', `sof/pipe-src-volume-playback.m4'))
+ifelse(SOF_ABI_VERSION_3_9_OR_GRT, `1',
+        define(`PIPE_VOLUME_PLAYBACK', `sof/pipe-volume-playback.m4'))
+
+#SSP1 setting for CML platform
+define(`SSP1_INDEX', 1)
+define(`SSP1_NAME', `SSP1-Codec')
+define(`SSP1_MCLK_RATE', `24000000')
+
+undefine(`SSP1_VALID_BITS')
+ifelse(SOF_ABI_VERSION_3_9_OR_GRT, `0',
+        define(`SSP1_VALID_BITS', `16'))
+ifelse(SOF_ABI_VERSION_3_9_OR_GRT, `1',
+        define(`SSP1_VALID_BITS', `24'))
+
+# playback DAI is SSP1 using 2 periods
+# Buffers use s16le format, with 48 frame per 1000us on core 0 with priority 0
+# With m/n divider available we can support 24 bit playback
+undefine(`SSP1_VALID_BITS_STR')
+ifelse(SOF_ABI_VERSION_3_9_OR_GRT, `0',
+        define(`SSP1_VALID_BITS_STR', `s16le'))
+ifelse(SOF_ABI_VERSION_3_9_OR_GRT, `1',
+        define(`SSP1_VALID_BITS_STR', `s24le'))
+
+# Fred: sof-cml-rt5682-max98357.m4 has variable rate
+#       sof-cml-rt1011-rt5682.m4 has fixed rate 48000
+ifelse(SOF_ABI_VERSION_3_9_OR_GRT, `0',
+        define(`SSP1_BCLK', `1500000'))
+ifelse(SOF_ABI_VERSION_3_9_OR_GRT, `1',
+        define(`SSP1_BCLK', `2304000'))
+
+ifelse(SOF_ABI_VERSION_3_9_OR_GRT, `0',
+        define(`SSP1_FSYNC', `46875'))
+ifelse(SOF_ABI_VERSION_3_9_OR_GRT, `1',
+        define(`SSP1_FSYNC', `48000'))
+
+
+#SSP 1 (ID: 6)
+#Use BCLK delay in SSP_CONFIG_DATA only on supporting version
+# Fred: sof-cml-rt5682-max98357.m4 uses 3 parameters
+#       sof-cml-rt1011-rt5682.m4 uses 6 or 3 parameters
```

Comment:
The macro is just outputting a text string, so number of args only matters to the target/user macro of this output.

Reply:
I handled different way between two files.

- SSP_CONFIG_DATA in sof-cml-rt5682-max98357.m4 has 3 parameters, so I used variable expansion.
- SSP_CONFIG_DATA in sof-cml-rt1011-rt5682.m4 has variable number of parameters. In this case I used entire parameter expansion, `SET_SSP1_CONFIG_DATA`